interfaces/tests/socket_transport_int/socket_transport_int.c

```diff
@@ -234,4 +253,160 @@ TEST_FUNCTION(send_and_receive_random_buffer_of_random_byte_succeeds)
     socket_transport_destroy(listen_socket);
 }
 
+static void wait_for_value(volatile_atomic int32_t* counter, int32_t target_value)
+{
+    int32_t value;
+    while ((value = interlocked_add(counter, 0)) != target_value)
+    {
+        (void)wait_on_address(counter, value, UINT32_MAX);
+    }
+}
+
+static int thread_worker_func(void* parameter)
+{
+    size_t i;
+    CHAOS_TEST_SOCKETS* chaos_test = parameter;
+
+    chaos_test->listen_socket = socket_transport_create_server();
+    ASSERT_IS_NOT_NULL(chaos_test->listen_socket);
+
+    ASSERT_ARE_EQUAL(int, 0, socket_transport_listen(chaos_test->listen_socket, g_port_num));
+
+    for (i = 0; i < CHAOS_THREAD_COUNT; i++)
+    {
+        ASSERT_ARE_EQUAL(int, 0, socket_transport_connect(chaos_test->client_socket_handles[i], "localhost", g_port_num, 10000));
+
+        chaos_test->incoming_socket_handles[i] = socket_transport_accept(chaos_test->listen_socket);
+
+        ASSERT_IS_NOT_NULL(chaos_test->incoming_socket_handles[i]);
+    }
+
+    (void)interlocked_increment(&chaos_test->API_create_result);
+    wake_by_address_single(&chaos_test->API_create_result);
+    return 0;
+}
+
+/* chaos test
+* This test will do the following:
+* create a bunch of client sockets to connect to binding socket every second
+* create a thread to listen and connect
+* send and receive random buffers of data per client
+* Disconnect a random incoming socket
+* Test if the disconnected socket receives the data from the client socket
+*/
+TEST_FUNCTION(socket_transport_chaos_knight_test)
+{
+    CHAOS_TEST_SOCKETS chaos_knight_test;
+
+    for (size_t i = 0; i < CHAOS_THREAD_COUNT; i++)
+    {
+        chaos_knight_test.client_socket_handles[i] = socket_transport_create_client();
+
+        ASSERT_IS_NOT_NULL(chaos_knight_test.client_socket_handles[i]);
+
+    }
+
+    THREAD_HANDLE thread;
+
+    interlocked_exchange(&chaos_knight_test.API_create_result, 0);
+    ASSERT_ARE_EQUAL(THREADAPI_RESULT, THREADAPI_OK, ThreadAPI_Create(&thread, thread_worker_func, &chaos_knight_test));
+    wait_for_value(&chaos_knight_test.API_create_result, 1);
```

Comment:
After looking at this, we don't need to have this in a thread since we're waiting for the values.  Just change thread_worker_func to be a normal function and call it normally.